grin/src/Pipeline/Pipeline.hs

```diff
@@ -623,6 +633,11 @@ pureEval = do
 
 printGrinM :: (Doc -> Doc) -> PipelineM ()
 printGrinM color = do
+  Program exts defs <- use psExp
+  pipelineLog $ showWide $ color $ prettyWithExternals exts $ Program [] defs
```

Comment:
A dedicated `prettyPrintProgram` with options such as highlight externals, render external would be better in the Pretty module, than let the client code combine such a functionality. It introduces an unnecessary noise. Also using the parameters for the `prettyPrintProgram`  could be reused and `printGrinM` and `fullPrintGrinM` should be merged.

Reply:
Done.